AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -125,6 +125,7 @@ def __init__(self, queue_info, app):
         self.task_retry_limit = retry_params['task_retry_limit']
 
     self.validate_config()
+    self.prepared_statements = {}
```

Comment:
It shouldn't be garbage collected. It's trading a bit of memory usage for better CPU performance. It would definitely be a leak if the items in this dict are no longer useful. However, there can only ever be 5 items in this dictionary (because only 5 prepared statements are ever cached per queue), and they are used any time the relevant operation is performed.

The keys are basically templates for common operations, and if the the statement (template) has already been prepared, the function can use that instead of adding a new item to the dict.